### PR TITLE
[3.0.x] WFS sample: further fixes

### PR DESCRIPTION
**Pull request by janusw**
* follow-up to #1467
* make labels show up
* make WFS layer slightly transparent

### CODE REVIEW

File: Samples/Mapsui.Samples.Common/Maps/Data/WfsSample.cs [hunk 1]
```diff
@@ -56,7 +56,7 @@ private static ILayer CreateWfsLayer(WFSProvider provider)
         {
             return new Layer(layerName)
             {
-                Style = new VectorStyle {Fill = new Brush {Color = Color.Red}},
+                Style = new VectorStyle { Fill = new Brush { Color = Color.FromArgb(192, 255, 0, 0) } },
```
Comment (inforithmics) [inline Samples/Mapsui.Samples.Common/Maps/Data/WfsSample.cs:59]: Good improvement now you can see through

**Review by inforithmics — APPROVED** [review]

**Comment by JBou** [comment]
Is it possible to make the labels visible without zooming in? Can you otherwise explain me why they are hidden after a specific zoom level. Is it even possible to control this?

**Comment by janusw** [comment]
> Is it possible to make the labels visible without zooming in? Can you otherwise explain me why they are hidden after a specific zoom level. Is it even possible to control this?

I'm not sure, but I would assume the labels are hidden if the features become to small and/or the labels would be overlapping with each other. So it looks less like a bug and more like a feature to me ;)

I don't know if there is a way to control it, but we can check the code.

**Comment by JBou** [comment]
No problem, I don't think it's a bug either ;)
Just wanted to point out that new users may not easily see the labels because they need to zoom, it may not be user-friendly. But it's only my opinion, if there was an easy fix we can fix it, otherwise we can just leave it as it is. It's only a sample, and if someone is interested, he willl find it.

**Comment by janusw** [comment]
> Just wanted to point out that new users may not easily see the labels because they need to zoom, it may not be user-friendly.

Maybe an easy way to work around this problem would be to set the bounding box on startup such that the labels are visible by default?

**Comment by janusw** [comment]
> > Just wanted to point out that new users may not easily see the labels because they need to zoom, it may not be user-friendly.
> 
> Maybe an easy way to work around this problem would be to set the bounding box on startup such that the labels are visible by default?

Did that in 0677086371443136a74478f24e4ca4a54ac69613, which makes the labels show up in the inital view on MacOS. Ok with you?

I'll note that setting the bounding box does not seem to work on Android and iOS, where the map starts up with a whole-world view. No idea why. However, I can see the labels on both platforms when zooming in.

Performance-wise it seems that Android is much better than iOS (and MacOS): The app freezes shortly when selecting the WFS sample, but after that the navigation of the map is quite performant (much more than on iOS). One should have a look into the performance issues later.

**Comment by pauldendulk** [comment]
> I'll note that setting the bounding box does not seem to work on Android and iOS, where the map starts up with a whole-world view. No idea why. However, I can see the labels on both platforms when zooming in.

The MapControl.Map.Home method is intended for this scenario (Home may not be a good name. DefaultViewport might be better). There is a nasty problem here, when starting an app several things get initialized on distinct points in time. The size of the app is determined while the app is loaded, the screen sometimes resizes after it is shown for the first time, and the data is loaded asynchronously. At some point, when the screen size is determined and the relevant data is loaded you want to set the viewport. The idea is that the MapControl calls the Home method at that point. But even the Home method might not work perfectly.

**Comment by pauldendulk** [comment]
> Performance-wise it seems that Android is much better than iOS (and MacOS): The app freezes shortly when selecting the WFS sample, but after that the navigation of the map is quite performant (much more than on iOS). One should have a look into the performance issues later.

This might be related to the FetchWorker and threads. I said something about this in the developer chat. This could be improved and I want to make an issue for it.

**Comment by JBou** [comment]
As stated on [this page](https://geoportal.buergernetz.bz.it/geodatendienste.asp), they are currently moving to a new infrastructure.
I have found the following new wfs service with the same feature:

`https://geoservices1.civis.bz.it/geoserver/p_bz-AdministrativeUnits/ows?Service=WFS`
Namespace: `p_bz-AdministrativeUnits`
Layername: `Districts`
Label: `NAME_DE`

I can test it out tomorrow, I'm not at the pc atm.
Once they have moved, it should be more stable. Bit I don't know if this will be the final wfs service url. Otherwise we can also move to another service.

**Comment by pauldendulk** [comment]
Is there no option in the xml parse logic to make it work with any namespace? 

btw, the implementation uses an older xml parser, XmlDocument is a newer one, perhaps it is easier to make it work with namespaces. It might be a fairly big rewrite though.

**Comment by janusw** [comment]
> Is there no option in the xml parse logic to make it work with any namespace?

Well, for the namespace alone this could work in principle. But you anyway need to choose which of the many layers you want to show (which is determined by the combination of namespace and layer name).


> btw, the implementation uses an older xml parser, XmlDocument is a newer one, perhaps it is easier to make it work with namespaces. It might be a fairly big rewrite though.

Might be worth it. Right now the WFS implementation seems to be a bit shaky and unstable (throwing various errors and warnings in a non-deterministic way, and sometimes also parts of the rendering seem to be missing).

**Pull request merged**